test/jdk/java/awt/im/JTextFieldTest.java

```diff
@@ -23,43 +23,53 @@
 
 /*
  * @test
- * @bug  4226191
+ * @bug 4226191
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
  * @summary  Verify that Lightweight text components (like swing JTextField)
  *           work correctly with IM when there is an uneditable peered
  *           TextField/TextArea in the same parent Frame
- * @author xueming.shen@eng
- * @run applet/manual=yesno JTextFieldTest.html
+ * @run main JTextFieldTest
  */
 
-import java.awt.*;
-import java.awt.event.*;
-import java.applet.*;
-import javax.swing.*;
+import java.awt.FlowLayout;
+import java.awt.TextField;
 
-public class JTextFieldTest extends Applet implements ActionListener {
+import javax.swing.JFrame;
+import javax.swing.JTextField;
+import javax.swing.WindowConstants;
 
-    TextField  tf1;
-    JTextField tf2;
+public class JTextFieldTest {
+    private static JFrame frame;
+    private static final String INSTRUCTIONS =
+            """
+             Please run this test case under Solaris cjk locale with inputmethod
+             support, if you could input Chinese/Japanese/Korean in the swing
```

Comment:
The scenario here should not be specific to Solaris.
I think you should rephrase the first sentence to
"Please run this test in a CJK (Chinese/Japanese/Korean) locale with input method support".

I some times wonder how well these tests that say "please change your environment" actually get properly tested.
It is perhaps possible to do change desktop locale support on the fly but by the time
you read the instructions the test is already running.
That introduces another wrinkle. Maybe you need to restart at least this test.
And I am not 100% sure that jtreg passes through locale vars. I hope it does.

Reply:
Updated the instructions.